dynawo/sources/Solvers/SolverSIM/DYNSolverSIM.cpp

```diff
@@ -341,6 +343,14 @@ void SolverSIM::solveStep(double /*tAim*/, double& tNxt) {
   } while (redoStep);
 
   updateTimeStep(tNxt);
+
+  if (std::abs(tSolve_ - tNxt) < minimalAcceptableStep_) {
+    ++nbLastTimeSimulated_;
+    if (nbLastTimeSimulated_ > maximumNumberSlowStepIncrease_)
+      throw DYNError(Error::SOLVER_ALGO, SlowStepIncrease, maximumNumberSlowStepIncrease_, minimalAcceptableStep_);
+  } else {
+    nbLastTimeSimulated_ = 0;
+  }
```

Comment:
Could we include it in the updateTimeStep method (or any other one)? so that it doesn't "pollute" in a way the understanding at first glance of the solver algorithm...

If you disagree, just ignore the remark ;)

Reply:
done